src/languages/latex.js

```diff
@@ -105,7 +105,7 @@ export default function(hljs) {
   };
   const MAGIC_COMMENT = {
     className: 'meta',
-    begin: '% !TeX',
+    begin: /% ?!(?:T[eE]X|tex|BIB|bib)/,
```

Comment:
```suggestion
    begin: /% ?!(T[eE]X|tex|BIB|bib)/,
```

We usually don't care about capture or not and write simpler regex... I don't see any reason this needs to be non-capture?  I've also found cases where capture is much faster (which makes no sense to me, but is what it is)... so I say we stick with momentum and simpler.

Reply:
Interesting. As a rule, I use non-capturing groups in regexes whenever I don't need the capturing. Maybe I should overthink this..

Reply:
That's probably a good general rule for many cases.  In other systems (Prism.js, etc) this matters a LOT... but for our scope it's kind of never mattered and it's a bit simpler to not think about it.